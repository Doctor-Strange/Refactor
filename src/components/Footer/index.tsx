/* tslint:disable */
import * as React from "react";
import { useState, useEffect } from "react";
import Link from "next/link";
import Router from "next/router";

const Footer = () => {
  const [hider, setHider] = useState(false);

  useEffect(() => {
    if (Router.route === "/checkout") setHider(true);
  }, []);
  return (
    <footer>
      <div id="toTop" />
      <div className={["plus_border", hider ? "removeFooter" : ""].join(" ")}>
        <div className="container" dir="rtl">
          <div className="row">
            <div
              className="col-lg-3 col-md-6 col-sm-6"
              style={{ borderLeft: "1px solid #e8e8e8" }}
            >
              <div className="collapse show" id="collapse_ft_2">
                <ul className="links">
                  <li>
                    <Link href="/about-us">
                      <a>درباره ما</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/otoli">
                      <a>اتولی چگونه کار می‌کند؟</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/faq">
                      <a>سوالات متداول</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/our-policies">
                      <a>شرایط و قوانین استفاده</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/insurance-policies">
                      <a>شرایط و پوشش‌های بیمه</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/rent">
                      <a>اجاره ماشین</a>
                    </Link>
                  </li>
                  <li>
                    <Link href="/site-map">
                      <a>نقشه سایت</a>
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
            <div className="col-lg-3 col-md-6 col-sm-6 jus">
              <div className="follow_us">
                <h3 style={{ color: "#929292" }}>
                  ما را در شبکه‌های اجتماعی دنبال کنید:
                </h3>
                {/* <Icon name='facebook' /> */}
                <div className="iconContainer">
                  <a
                    style={{ margin: 0 }}
                    target="_black"
                    href="https://www.instagram.com/otoli_net/"
                  >
                    {/* <Icon size="large" name="instagram" style={{ margin: 0 }} /> */}
                  </a>
                  {/* <Icon name='google plus' /> */}
                  {/* <Icon name='vk' /> */}

                  <a target="_black" href="https://twitter.com/otoli_net">
                    {/* <Icon size="large" name="twitter" /> */}
                  </a>
                  <a target="_black" href="https://t.me/otoli_net">
                    {/* <Icon size="large" name="telegram" /> */}
                  </a>
                </div>
                <p
                  style={{
                    color: "#929292",
                    marginTop: "16px",
                    fontWeight: 500
                  }}
                >
                  شماره تماس: <a href="tel:02188567759">۰۲۱۸۸۵۶۷۷۵۹</a>
                </p>
              </div>
            </div>
          </div>

          <div className="row">
            <div className="col-lg-12">
              <span className="ver">2.0.0</span>
              <Link href="/site-map">
                <a
                  style={{
                    color: "transparent",
                    display: "inline-block"
                  }}
                >
                  site map
                </a>
              </Link>
              <p>تمامی حقوق برای وب‌سایت اتولی محفوظ است.</p>
            </div>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
