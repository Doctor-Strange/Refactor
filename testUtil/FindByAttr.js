

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
