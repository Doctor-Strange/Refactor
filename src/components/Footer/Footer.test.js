import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import Footer from "./";

const setup = (props) => {
  return mount(<Footer {...props} />);
};

describe("/component/footer", () => {
  test("render without crash ", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
describe("/component/footer", () => {
    const wrapper = setup({ hide: true });
  test("Hide footer", () => {
    const footer = FindByAttr(wrapper, "footer");
    expect(footer.props().className).toEqual("hide_footer");
  });
});
