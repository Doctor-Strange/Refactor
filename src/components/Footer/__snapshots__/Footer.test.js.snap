// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/component/footer render without crash  1`] = `ReactWrapper {}`;
